File: LeetCode/python/921.py
class Solution(object):
    def minAddToMakeValid(self, S):
        """
        :type S: str
        :rtype: int
        """
        parentheses = []
        for c in S:
            if c == '(':
                parentheses.append(c)
            else:
                if len(parentheses) == 0 or parentheses[-1] != '(':
                    parentheses.append(c)
                else:
                    parentheses.pop()
        return len(parentheses)

File: LeetCode/cpp/921.cpp
class Solution {
public:
    int minAddToMakeValid(string S) {
        stack<char> parentheses;
        for (auto c:S) {
            if (c == '(')
                parentheses.push(c);
            else {
                if (!parentheses.empty() and parentheses.top() == '(')
                    parentheses.pop();
                else
                    parentheses.push(c);
            }
        }
        return parentheses.size();
    }
};
